module-services/service-desktop/USBSecurityModel.cpp

```diff
@@ -50,9 +51,20 @@ namespace sdesktop
             settings->getValue(settings::SystemProperties::eulaAccepted, settings::SettingsScope::Global));
     }
 
+    auto USBSecurityModel::isBatteryLevelCritical() const -> bool
+    {
+        using LevelState = Store::Battery::LevelState;
+
+        return Store::Battery::get().levelState == LevelState::CriticalCharging or
+               Store::Battery::get().levelState == LevelState::CriticalNotCharging;
+    }
+
     auto USBSecurityModel::getEndpointSecurity() const -> endpointSecurity_t
     {
-        if (!isEulaAccepted()) {
+        if (isBatteryLevelCritical()) {
+            return {EndpointSecurity::Block, BlockReason::BatteryCriticalLevel};
+        }
+        else if (!isEulaAccepted()) {
```

Comment:
If each branch returns early, it's not required to use else/if. In this case simply stacking ifs will be more readable solution imho.
```suggestion
        if (isBatteryLevelCritical()) {
            return {EndpointSecurity::Block, BlockReason::BatteryCriticalLevel};
        }
        if (!isEulaAccepted()) {
```